File: 1_encapsulation/1_python/3_range.py
class Sensor:
    def __init__(self, sensor_id, location):
        self.__sensor_id = sensor_id  # 私有成员
        self._location = location  # 受保护成员

    def read_value(self):
        return "Reading sensor value..."

    def get_status(self):
        return "Sensor status OK"

    # 公有方法
    def get_sensor_id(self):
        return self.__sensor_id

# 创建对象并访问成员
temp_sensor = Sensor(sensor_id="T1", location="Living Room")
print(temp_sensor.read_value())
print(temp_sensor.get_status())
print(temp_sensor.get_sensor_id())

File: 1_encapsulation/2_cpp/3_range.cc
#include <iostream>
#include <string>

class Sensor
{
private:
    std::string sensor_id; // 私有成员
protected:
    std::string location; // 受保护成员
public:
    Sensor(const std::string &id, const std::string &loc) : sensor_id(id), location(loc) {}
    std::string read_value()
    {
        return "Reading sensor value...";
    }
    std::string get_status()
    {
        return "Sensor status OK";
    }
    // 公有方法
    std::string get_sensor_id()
    {
        return sensor_id;
    }
};

int main()
{
    Sensor temp_sensor("T1", "Living Room");
    std::cout << temp_sensor.read_value() << std::endl;
    std::cout << temp_sensor.get_status() << std::endl;
    std::cout << temp_sensor.get_sensor_id() << std::endl;
    std::cout << temp_sensor.sensor_id << std::endl; // error
    return 0;
}
